enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,205 @@
+* When WMCO is upgraded from 5.x.z to 6.x.z, the runtime on all existing Windows nodes will be migrated to
+  containerd.  
+
+### Non-Goals
+
+* De-configuring Docker runtime is not part of this enhancement.
+* Allowing the user to choose a  container runtime migrating back to Docker from containerd is not supported.
+*  Downgrade is not supported.
+
+## Proposal
+
+To make containerd the default runtime, containerd should be installed as a Windows
+service before kubelet gets installed. While starting kubelet, parameters `container-runtime` and
+`container-runtime-endpoint`should be configured. The remaining WMCO workflow has no impact other than the proposed
+changes. MachineSet and BYOH (Bring Your Own Host) upgrade details are discussed in the upgrade section.
+There is no difference between MachineSet and BYOH on how we enable containerd as the default runtime.
+
+Golden Image:
+
+In MachineSet use case, the Windows image has preinstalled Docker runtime. Users can create a new Windows
+image excluding Docker runtime or continue to use the same Windows image. Both images will be supported.
+
+Containerd Migration plan:
+
+Containerd will be the default runtime in OpenShift 4.11. For development, we plan to introduce a
+WMCO CLI option which allows us to choose the runtime. This CLI option will be used in the
+release of the WMCO community operator for OpenShift 4.10 to allow customers to try out Windows
+nodes with containerd. Customers can use 4.10 community WMCO to create new clusters or upgrade
+existing clusters.
+   
+### User Stories
+
+Stories can be found within the [Windows Containers: containerd epic](https://issues.redhat.com/browse/WINC-505)
+
+### Justification
+We have explored containerd, CRIO, and Mirantis dockershim as an alternate runtime. After looking into business and
+engineering details, we conclude containerd stands out to be the best solution for WMCO. 
+
+Advantage of using containerd:
+* Containerd is widely adopted and supported by the open-source community.
+* With minimal effort, we get new features and bug fixes from the community and enterprise.
+* Better CPU performance and memory usage compared with CRIO and Docker
+* Microsoft is a major contributor in Windows containerd development and moved to containerd as the default
+runtime for their Kubernetes offerings.
+* Most of the Windows-supported Kubernetes orchestrators already moved to containerd.
+
+Disadvantage of using other runtimes:
+* Mirantis supported dockershim will continue to use Docker. Using Docker as runtime has an impact on memory
+and CPU performance.
+* From the business point of view subscribing to Mirantis for dockershim support is not an ideal situation as Mirantis
+is our key competitor.
+* The amount of time and engineering efforts involves in making CRI-O to work for Windows is huge. This doesn't
+do business justification due to the high cost and time.
+* Less traction from the community to develop CRIO for Windows.
```

Comment:
This may be an understatement :)

Reply:
Then we would have already asked about it or some traction would have come from the community. I can remove it if you still feel understatement :)
Take containerd as an example. It started with Linux and Windows support eventually came in with community support.

Reply:
No need to remove it, i think its a good bullet point.
I think there is close to zero community drive to develop CRI-O for Windows.
Much, much less than containerd at least.